src/main/java/frc/robot/subsystems/vision/VisionEstimation.java

```diff
@@ -0,0 +1,159 @@
+
+        allNotifier.setName("runAll");
+        allNotifier.startPeriodic(0.02);
+
+        // prevTime.set(Timer.getFPGATimestamp());
+        this.swerveSubsystem.getPose();
+    }
+
+    @Override
+    public void periodic() {
+        if (VisionConstants.USE_VISION) {
+            estimatorChecker(frontRightEstimator);
+            estimatorChecker(frontLeftEstimator);
+            estimatorChecker(backRightEstimator);
+            estimatorChecker(backLeftEstimator);
+        } else {
+            allNotifier.close();
+        }
+
+        // Set the pose on the dashboard
+        var dashboardPose = swerveSubsystem.getPose();
+        if (originPosition == kRedAllianceWallRightSide) {
+            // Flip the pose when red, since the dashboard field photo cannot be rotated
+            dashboardPose = flipAlliance(dashboardPose);
+        }
+    }
+
+    /**
+    * Transforms a pose to the opposite alliance's coordinate system. (0,0) is
+    * always on the right corner of your
+    * alliance wall, the field elements are at different coordinates
+    * for each alliance.
+    * 
+    * @param poseToFlip pose to transform to the other alliance
+    * @return pose relative to the other alliance's coordinate system
+    */
+    private Pose2d flipAlliance(Pose2d poseToFlip) {
+        return poseToFlip.relativeTo(new Pose2d(
+                new Translation2d(FieldConstants.length, FieldConstants.width),
+                new Rotation2d(Math.PI)));
+    }
+
+    // public void addTrajectory(PathPlannerTrajectory traj) {
+    //   field2d.getObject("Trajectory").setTrajectory(traj);
+    // }
+
+    private Matrix<N3, N1> confidenceCalculator(EstimatedRobotPose estimation) {
+        double smallestDistance = Double.POSITIVE_INFINITY;
+        for (var target : estimation.targetsUsed) {
+            var target3D = target.getBestCameraToTarget();
+            var distance = Math
+                    .sqrt(Math.pow(target3D.getX(), 2) + Math.pow(target3D.getY(), 2) + Math.pow(target3D.getZ(), 2));
+            if (distance < smallestDistance)
+                smallestDistance = distance;
+        }
+        double poseAmbiguityFactor = estimation.targetsUsed.size() != 1
+                ? 1
+                : Math.max(
+                        1,
+                        (estimation.targetsUsed.get(0).getPoseAmbiguity()
+                                + VisionConstants.POSE_AMBIGUITY_SHIFTER)
+                                * VisionConstants.POSE_AMBIGUITY_MULTIPLIER);
+        double confidenceMultiplier = Math.max(
+                1,
+                (Math.max(
+                        1,
+                        Math.max(0, smallestDistance - VisionConstants.NOISY_DISTANCE_METERS)
+                                * VisionConstants.DISTANCE_WEIGHT)
+                        * poseAmbiguityFactor)
+                        / (1
+                                + ((estimation.targetsUsed.size() - 1) * VisionConstants.TAG_PRESENCE_WEIGHT)));
```

Comment:
there's a bit too much going on in these ternaries; I'd suggest pulling some computed numbers out to variables.

Reply:
ok